### ISSUE TITLE
[LowMemory] Issue when XF app wakes after resource-intensive game on Android

### ISSUE THREAD

**Issue by JTOne123** [MvvmCross/MvvmCross#2430]
### Steps to reproduce :scroll:

1) Deploy Playground.Forms.Droid to device (On Nexus 5 reproduce every time, do not use Samsung)

2) Run high load resource-intensive 3D game and play some time OR MemoryPump (https://play.google.com/store/apps/details?id=com.fabsimilian.memorypump&hl=en) [Press FILL ALL MEM]

3) After low memory, our app will be killed

4) Press Square and choose our app in recent

5) And we have a WHITE SCREEN (no crash, no errors, nothing in logs)

Here's log, that's all, nothing else:

```
12-01 15:20:12.042	LGE Nexus 5	Info	4209	mono-stdout	mvx : Diagnostic :   0,45 Setup: PlatformServices start, System.Object[]
12-01 15:20:14.709	LGE Nexus 5	Info	4209	mono-stdout	mvx : Diagnostic :   3,19 Null Extras seen on Intent when creating ViewModel - have you tried to navigate to an MvvmCross View directly? Will try direct load, System.Object[]
12-01 15:20:14.441	LGE Nexus 5	Info	4209	mono-stdout	mvx : Diagnostic :   2,92 Setup: Secondary end, System.Object[]
12-01 15:20:13.934	LGE Nexus 5	Info	4209	mono-stdout	mvx : Diagnostic :   2,41 Setup: LastChance start, System.Object[]
12-01 15:20:13.932	LGE Nexus 5	Info	4209	mono-stdout	mvx : Diagnostic :   2,41 Setup: InpcInterception start, System.Object[]
12-01 15:20:13.931	LGE Nexus 5	Info	4209	mono-stdout	mvx : Diagnostic :   2,41 Setup: NavigationSerializer start, System.Object[]
12-01 15:20:13.931	LGE Nexus 5	Info	4209	mono-stdout	mvx : Diagnostic :   2,41 Setup: InpcInterception start, System.Object[]
12-01 15:20:13.928	LGE Nexus 5	Info	4209	mono-stdout	mvx : Diagnostic :   2,41 Setup: CommandCollectionBuilder start, System.Object[]
12-01 15:20:13.868	LGE Nexus 5	Info	4209	mono-stdout	mvx : Diagnostic :   2,35 Setup: Views start, System.Object[]
12-01 15:20:12.698	LGE Nexus 5	Info	4209	mono-stdout	mvx : Diagnostic :   1,18 Setup: ViewDispatcher start, System.Object[]
12-01 15:20:12.694	LGE Nexus 5	Info	4209	mono-stdout	mvx : Diagnostic :   1,17 Setup: ViewsContainer start, System.Object[]
12-01 15:20:12.679	LGE Nexus 5	Info	4209	mono-stdout	mvx : Diagnostic :   1,16 Setup: ViewModelTypeFinder start, System.Object[]
12-01 15:20:12.357	LGE Nexus 5	Info	4209	mono-stdout	mvx : Warning :   0,84 Language file could not be loaded for {0}.{1} - {2}, System.Object[]
12-01 15:20:12.357	LGE Nexus 5	Info	4209	mono-stdout	mvx : Warning :   0,84 Language file could not be loaded for {0}.{1} - {2}, System.Object[]
12-01 15:20:12.356	LGE Nexus 5	Info	4209	mono-stdout	mvx : Warning :   0,84 Language file could not be loaded for {0}.{1} - {2}, System.Object[]
12-01 15:20:12.356	LGE Nexus 5	Info	4209	mono-stdout	mvx : Warning :   0,84 Language file could not be loaded for {0}.{1} - {2}, System.Object[]
12-01 15:20:12.355	LGE Nexus 5	Info	4209	mono-stdout	mvx : Warning :   0,83 Language file could not be loaded for {0}.{1} - {2}, System.Object[]
12-01 15:20:12.355	LGE Nexus 5	Info	4209	mono-stdout	mvx : Warning :   0,83 Language file could not be loaded for {0}.{1} - {2}, System.Object[]
12-01 15:20:12.355	LGE Nexus 5	Info	4209	mono-stdout	mvx : Warning :   0,84 Language file could not be loaded for {0}.{1} - {2}, System.Object[]
12-01 15:20:12.354	LGE Nexus 5	Info	4209	mono-stdout	mvx : Warning :   0,83 Language file could not be loaded for {0}.{1} - {2}, System.Object[]
12-01 15:20:12.354	LGE Nexus 5	Info	4209	mono-stdout	mvx : Warning :   0,83 Language file could not be loaded for {0}.{1} - {2}, System.Object[]
12-01 15:20:12.353	LGE Nexus 5	Info	4209	mono-stdout	mvx : Warning :   0,83 Language file could not be loaded for {0}.{1} - {2}, System.Object[]
12-01 15:20:12.353	LGE Nexus 5	Info	4209	mono-stdout	mvx : Warning :   0,83 Language file could not be loaded for {0}.{1} - {2}, System.Object[]
12-01 15:20:12.353	LGE Nexus 5	Info	4209	mono-stdout	mvx : Warning :   0,83 Language file could not be loaded for {0}.{1} - {2}, System.Object[]
12-01 15:20:12.352	LGE Nexus 5	Info	4209	mono-stdout	mvx : Warning :   0,83 Language file could not be loaded for {0}.{1} - {2}, System.Object[]
12-01 15:20:12.352	LGE Nexus 5	Info	4209	mono-stdout	mvx : Warning :   0,83 Language file could not be loaded for {0}.{1} - {2}, System.Object[]
12-01 15:20:12.351	LGE Nexus 5	Info	4209	mono-stdout	mvx : Warning :   0,83 Language file could not be loaded for {0}.{1} - {2}, System.Object[]
12-01 15:20:12.351	LGE Nexus 5	Info	4209	mono-stdout	mvx : Warning :   0,83 Language file could not be loaded for {0}.{1} - {2}, System.Object[]
12-01 15:20:12.350	LGE Nexus 5	Info	4209	mono-stdout	mvx : Warning :   0,83 Language file could not be loaded for {0}.{1} - {2}, System.Object[]
12-01 15:20:12.350	LGE Nexus 5	Info	4209	mono-stdout	mvx : Warning :   0,83 Language file could not be loaded for {0}.{1} - {2}, System.Object[]
12-01 15:20:12.350	LGE Nexus 5	Info	4209	mono-stdout	mvx : Warning :   0,83 Language file could not be loaded for {0}.{1} - {2}, System.Object[]
12-01 15:20:12.349	LGE Nexus 5	Info	4209	mono-stdout	mvx : Warning :   0,83 Language file could not be loaded for {0}.{1} - {2}, System.Object[]
12-01 15:20:12.349	LGE Nexus 5	Info	4209	mono-stdout	mvx : Warning :   0,83 Language file could not be loaded for {0}.{1} - {2}, System.Object[]
12-01 15:20:12.348	LGE Nexus 5	Info	4209	mono-stdout	mvx : Warning :   0,83 Language file could not be loaded for {0}.{1} - {2}, System.Object[]
12-01 15:20:12.348	LGE Nexus 5	Info	4209	mono-stdout	mvx : Warning :   0,83 Language file could not be loaded for {0}.{1} - {2}, System.Object[]
12-01 15:20:12.348	LGE Nexus 5	Info	4209	mono-stdout	mvx : Warning :   0,83 Language file could not be loaded for {0}.{1} - {2}, System.Object[]
12-01 15:20:12.347	LGE Nexus 5	Info	4209	mono-stdout	mvx : Warning :   0,83 Language file could not be loaded for {0}.{1} - {2}, System.Object[]
12-01 15:20:12.347	LGE Nexus 5	Info	4209	mono-stdout	mvx : Warning :   0,83 Language file could not be loaded for {0}.{1} - {2}, System.Object[]
12-01 15:20:12.347	LGE Nexus 5	Info	4209	mono-stdout	mvx : Warning :   0,83 Language file could not be loaded for {0}.{1} - {2}, System.Object[]
12-01 15:20:12.346	LGE Nexus 5	Info	4209	mono-stdout	mvx : Warning :   0,83 Language file could not be loaded for {0}.{1} - {2}, System.Object[]
12-01 15:20:12.346	LGE Nexus 5	Info	4209	mono-stdout	mvx : Warning :   0,83 Language file could not be loaded for {0}.{1} - {2}, System.Object[]
12-01 15:20:12.345	LGE Nexus 5	Info	4209	mono-stdout	mvx : Warning :   0,82 Language file could not be loaded for {0}.{1} - {2}, System.Object[]
12-01 15:20:12.344	LGE Nexus 5	Info	4209	mono-stdout	mvx : Warning :   0,82 Language file could not be loaded for {0}.{1} - {2}, System.Object[]
12-01 15:20:12.199	LGE Nexus 5	Info	4209	mono-stdout	mvx : Diagnostic :   0,68 Setup: App start, System.Object[]
12-01 15:20:12.187	LGE Nexus 5	Info	4209	mono-stdout	mvx : Diagnostic :   0,67 Setup: Load navigation routes, System.Object[]
12-01 15:20:12.175	LGE Nexus 5	Info	4209	mono-stdout	mvx : Diagnostic :   0,65 Setup: NavigationService, System.Object[]
12-01 15:20:12.174	LGE Nexus 5	Info	4209	mono-stdout	mvx : Diagnostic :   0,65 Setup: Create App, System.Object[]
12-01 15:20:12.170	LGE Nexus 5	Info	4209	mono-stdout	mvx : Diagnostic :   0,65 Setup: PluginManagerFramework start, System.Object[]
12-01 15:20:12.169	LGE Nexus 5	Info	4209	mono-stdout	mvx : Diagnostic :   0,65 Setup: CommandHelper start, System.Object[]
12-01 15:20:12.154	LGE Nexus 5	Info	4209	mono-stdout	mvx : Diagnostic :   0,63 Setup: StringToTypeParser start, System.Object[]
12-01 15:20:12.058	LGE Nexus 5	Info	4209	mono-stdout	mvx : Diagnostic :   0,54 Setup: Bootstrap actions, System.Object[]
12-01 15:20:12.054	LGE Nexus 5	Info	4209	mono-stdout	mvx : Diagnostic :   0,53 Setup: Singleton Cache start, System.Object[]
12-01 15:20:12.053	LGE Nexus 5	Info	4209	mono-stdout	mvx : Diagnostic :   0,53 Setup: MvvmCross settings start, System.Object[]
```
---
This issue reproduces on any version of mvvmcross (5.5.2 included).
---
XF without mvvmcross restore fine!
---

Also, I have found in MvxFormsAppCompatActivity, maybe it can help to produce some ideas...
```
        public virtual void InitializeForms(Bundle bundle)
        {

//After restoring FormsApplication.MainPage is null

            if (FormsApplication.MainPage != null)
            {
                global::Xamarin.Forms.Forms.Init(this, bundle, GetResourceAssembly());
                LoadApplication(FormsApplication);
            }
        }

```

### Expected behavior :thinking:
Tell us what should happen
App should be restored correctly

### Actual behavior :bug:
Tell us what happens instead
White screen

### Configuration :wrench:

**Version:** 5.5.2

**Platform:** 
- [ ] :iphone: iOS
- [x] :robot: Android
- [ ] :checkered_flag: WPF
- [ ] :earth_americas: UWP
- [ ] :apple: MacOS
- [ ] :tv: tvOS
- [x] :monkey: Xamarin.Forms

**Comment by martijn00** [comment]
This seems to be related to having a splashscreen with MvvmCross.

**Comment by JTOne123** [comment]
I guess so

**Comment by JTOne123** [comment]
I found a workaround for this issue (mvvmcross 5.7).
The root issue is https://github.com/MvvmCross/MvvmCross/issues/2261

```
    public class FormsApplicationActivity : MvxFormsAppCompatActivity
    {
        protected override void OnCreate(Bundle bundle)
        {
            ToolbarResource = Resource.Layout.Toolbar;
            TabLayoutResource = Resource.Layout.Tabbar;

            var wasInitialized = Xamarin.Forms.Forms.IsInitialized;

            base.OnCreate(bundle);

            if (!wasInitialized)
                InitWithoutSplash(bundle);
        }

        private void InitWithoutSplash(Bundle bundle)
        {
            //any additional initializations

            var starter = Mvx.Resolve<IMvxAppStart>();
            starter.Start();

            InitializeForms(bundle);
        }
}
```

**Comment by martijn00** [comment]
@JTOne123 This should be fixed in 6.0 then.
